src/s3/headbucket.cpp

```diff
@@ -0,0 +1,68 @@
+#include "./s3_api.hpp"
+#include "../authentication.hpp"
+#include "../bucket.hpp"
+#include "../common_routines.hpp"
+
+#include <irods/irods_exception.hpp>
+
+#include <fmt/format.h>
+
+namespace asio = boost::asio;
+namespace beast = boost::beast;
+namespace fs = irods::experimental::filesystem;
+
+static const std::string date_format{"{:%a, %d %b %Y %H:%M:%S GMT}"};
+
+boost::asio::awaitable<void> irods::s3::actions::handle_headbucket(
+    boost::asio::ip::tcp::socket& socket,
+    static_buffer_request_parser& parser,
+    const boost::urls::url_view& url)
+{
+    auto thing = irods::s3::get_connection();
+    beast::http::response<beast::http::empty_body> response;
+    response.result(beast::http::status::forbidden);
+    try {
+        auto irods_username = irods::s3::authentication::authenticates(*thing, parser, url);
+
+        if (!irods_username) {
+            response.result(beast::http::status::forbidden);
+        }
+        else {
+            // Reconnect to the iRODS server as the target user.
+            // The rodsadmin account from the config file will act as the proxy for the user.
+            thing = irods::s3::get_connection(irods_username);
+
+            fs::path path;
+            if (auto bucket = irods::s3::resolve_bucket(*thing, url.segments()); bucket.has_value()) {
+                path = bucket.value();
+                path = irods::s3::finish_path(path, url.segments());
+            }
+            else {
+                std::cout << "Failed to resolve bucket" << std::endl;
```

Comment:
Should we error out here, or is this being deferred to the `fs::exists` call below?

Reply:
I am not sure.  Just to be safe I changed this to return a `bad_request` in this case.